### ISSUE TITLE
Redirect if already OAuth authorized

### ISSUE THREAD

**Issue by moshthepitt** [onaio/onadata#1649]
It would be nice to not force a user to re-authorize an OAuth request that they had already successfully authorized.  We should instead just redirect them to the right (registered) callback url.

**Comment by WinnyTroy** [comment]
@ukanga @moshthepitt  @githengi @cKellyDesign
Django provides for a way to do this setting `skip_authorization` to True. This is for meant to be used esp by trusted applications authorizing using Onadata.

Setting this condition does not skip the `AuthorizationView` it only skips the HTML form display and immediately sends the user on their way to view the resources they requested. This is necessary, as the application still has to be "authorized" in the database (the tokens have to be created etc.), the user just doesn't get to keep hitting `Authorize` for this exchange of tokens to occur.

The current limitation found for this implementation is that it would restrict the user to only logging in to the client application with one Ona account(Ona as the oauth provider) as the API does not currently provide a logout endpoint. At the moment, logging out of the API is only possible using the Django Admin interface at the moment, which is intentionally not provided to the user.

**Comment by moshthepitt** [comment]
> Django provides for a way to do this setting skip_authorization to True. This is for meant to be used esp by trusted applications authorizing using Onadata.

How is this set @WinnyTroy ?

**Comment by WinnyTroy** [comment]
> How is this set @WinnyTroy ?

Following

**Comment by WinnyTroy** [comment]
> How is this set @WinnyTroy ?

@moshthepitt following Django OAuth Toolkit's documentation here: https://django-oauth-toolkit.readthedocs.io/en/latest/advanced_topics.html#skip-authorization-completely-for-trusted-applications

It can be set on the `Application model` either programmatically or using Django-Admin interface.
I was able to test it out locally using the Django-Admin interface for the GRP client application with help from @JohnMwashuma

**Comment by JohnMwashuma** [comment]
@WinnyTroy Found a solutions to the logout problem, the logout url for api accepts a `next` query param which can be the authorization url for an OAuth application.

Example:
`https://stage-api.ona.io/accounts/logout/?next=/o/authorize/......`

**Comment by WinnyTroy** [comment]
We were able to resolve this by setting the `skip_authorization` flag programmatically within Ona production servers for the grp and discover.io client applications.

Steps we followed to enable this setting were:
1. Ssh into the api servers.
2. Activate onadata virtualenv
3. Check into django shell by running `./manage.py  runserver`
4. import the Application server within django shell i.e `from oauth2_provider.models import Application`
5. Filter out the client application of choice using the client_id i.e `zebra_app  = Application.objects.filter(client_id='ZEBRA_PREVIEW')`
6. setting `skip_authorization` to True and saving the changes.

This should help apply to other client applications that authenticate with Onadata, if they wish to have this setting enabled.
